kernel/src/actions/deletion_vector.rs

```diff
@@ -229,97 +232,24 @@ pub(crate) fn treemap_to_bools(treemap: RoaringTreemap) -> Vec<bool> {
     }
 }
 
-#[derive(PartialEq)]
-#[repr(C)]
-pub enum VectorType {
-    Selection,
-    Deletion,
-}
-
-pub trait BitSet<V> {
-    fn set_bit(&mut self, idx: usize, value: impl Into<V>);
-}
-
-pub trait ContainerIterator<C, V>
-where
-    C: BitSet<V>,
-{
-    fn next_batch(&mut self, mut container: C, batch_size: usize) -> Option<C> {
-        self.next_batch_with(&mut container, batch_size)
-            .map(|_| container)
-    }
-    fn next_batch_with(&mut self, container: &mut C, batch_size: usize) -> Option<()>;
-}
-
-impl<C, V> BitSet<V> for C
-where
-    C: AsMut<[V]>,
-{
-    fn set_bit(&mut self, idx: usize, value: impl Into<V>) {
-        if let Some(i) = self.as_mut().get_mut(idx) {
-            *i = value.into();
-        } else {
-            tracing::info!("Index value in DV: {} is None", idx);
-        }
-    }
-}
-
+#[derive(Debug, Clone)]
 #[repr(C)]
-pub struct DeletionVectorBatch {
-    inner_iter: Peekable<IntoIter>,
-    position: usize,
-    total_rows: usize,
-    vector_type: VectorType,
-   
+pub struct DeletionVector {
+    inner: RoaringTreemap,
 }
 
-impl DeletionVectorBatch {
-    pub fn new(treemap: RoaringTreemap, total_rows: usize, vector_type: VectorType) -> Self {
-        let inner_iter = treemap.into_iter().peekable();
-        Self {
-            inner_iter,
-            position: 0,
-            total_rows,
-            vector_type,
-        }
-    }
-
-    pub fn full_vector(&mut self) -> Option<Vec<bool>> {
-        self.next_batch(vec![false; self.total_rows], self.total_rows)
-    }
-    
-    pub fn full_bit_vector(&mut self) -> Option<Vec<u8>> {
-        self.next_batch(vec![0; self.total_rows], self.total_rows)
+impl DeletionVector {
+    pub fn new(
+        fs_client: Arc<dyn FileSystemClient>,
+        parent: &Url,
+        descriptor: DeletionVectorDescriptor,
+    ) -> DeltaResult<Self> {
+        let inner = descriptor.read(fs_client, parent)?;
+        Ok(Self { inner })
     }
 
-    #[inline]
-    fn flip_value(&self) -> bool {
-        self.vector_type == VectorType::Deletion
-    }
-}
-
-impl<C, V> ContainerIterator<C, V> for DeletionVectorBatch
-where
-    C: BitSet<V> + AsMut<[V]>,
-    V: From<bool> + Copy,
-{
-    fn next_batch_with(&mut self, container: &mut C, batch_size: usize) -> Option<()> {
-        if self.position >= self.total_rows {
-            return None;
-        }
-        let slice_range = self.position as u64 + batch_size as u64;
-
-        while let Some(bit) = self.inner_iter.next() {
-            container.set_bit(bit as usize - self.position, self.flip_value());
-
-            if let Some(next_bit) = self.inner_iter.peek() {
-                if *next_bit > slice_range {
-                    break;
-                }
-            }
-        }
-        self.position += batch_size;
-        Some(())
+    pub fn row_indexes(self) -> Vec<u64> {
+        self.inner.into_iter().collect::<Vec<u64>>()
```

Comment:
I think the turbofish is unnecessary here, given that the return value is already strongly typed?
```suggestion
        self.inner.into_iter().collect()
```

Reply:
irrelevant now